python/ucxx/_lib_async/endpoint.py

```diff
@@ -51,26 +51,58 @@ def closed(self):
         """Is this endpoint closed?"""
         return self._ep is None or not self._ep.is_alive()
 
-    def abort(self):
+    def abort(self, period=10**10, max_attempts=1):
         """Close the communication immediately and abruptly.
         Useful in destructors or generators' ``finally`` blocks.
 
+        Despite the attempt to close communication immediately, in some
+        circumstances, notably when the parent worker is running a progress
+        thread, a maximum timeout may be specified for which the close operation
+        will wait. This can be particularly important for cases where the progress
+        thread might be attempting to acquire the GIL while the current
+        thread owns that resource.
+
         Notice, this functions doesn't signal the connected peer to close.
-        To do that, use `Endpoint.close()`
+        To do that, use `Endpoint.close()`.
+
+        Parameters
+        ----------
+        period: int
+            maximum period to wait (in ns) for internal endpoint operations
+            to complete, usually two operations (pre and post) are involved
+            thus the maximum perceived timeout should be multiplied by two.
+        max_attempts: int
+            maximum number of attempts to close endpoint, only applicable
+            if worker is running a progress thread and `period > 0`.
         """
         if self._ep is not None:
             logger.debug("Endpoint.abort(): 0x%x" % self.uid)
-            self._ep.close()
+            # Wait for a maximum of 10s
+            self._ep.close(period=period, max_attempts=max_attempts)
         self._ep = None
         self._ctx = None
 
-    async def close(self):
+    async def close(self, period=10**9, max_attempts=1):
```

Comment:
```suggestion
    async def close(self, period=10**10, max_attempts=1):
```

Reply:
I think you meant to write `10**10` here too @wence- , but please correct me if I missed something.

Reply:
I just copied the value that was here https://github.com/rapidsai/ucxx/pull/136/commits/1cef6d9b15162bea243a66e3a36f417d5f0a38a4, but 10s in both cases as a default seems fine.

Reply:
The value was `int(1e9)`, which is `10**10`. The other instance you actually wrote `10**10` but here you wrote `10**9` which I then assumed was accidental.

Also a side note: I started thinking about this 10s/1 attempt default and I'm not sure it makes sense. Perhaps a more sane default would be 3s/3 attempts. Perhaps it doesn't make sense to block for such a long period and giving up immediately, perhaps a shorter wait and multiple attempts will give it more room to unblock if something went wrong and retry. I'll leave it like this for now but will reconsider this in the future.

Reply:
`1e9` is `10**9` (at least last time I looked :P) but the other one was `10e9` (which is `10**10`). But this is an aside. I too think it might make sense to have a retry setup, but not sure.

Reply:
> `1e9` is `10**9` (at least last time I looked :P) but the other one was `10e9` (which is `10**10`).

Ha! True, it was my typo, twice! I meant to write `10e9` here _and_ in the code where you just corrected me about. 🤦